browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -243,6 +243,13 @@ int OnBeforeURLRequest_AdBlockTPPreWork(const ResponseCallback& next_callback,
     return net::OK;
   }
 
+  // Requests for main frames are handled by DomainBlockNavigationThrottle,
+  // which can display a custom interstitial with an option to proceed if a
+  // block is made. We don't need to check these twice.
+  if (ctx->resource_type == blink::mojom::ResourceType::kMainFrame) {
```

Comment:
can we add a test?

Reply:
Added in https://github.com/brave/brave-core/pull/8402/commits/7d51927a0dad2211f2a07ff26b36afd0ce0ef65e